### PR TITLE
Check links

### PR DESCRIPTION
**Pull request by nanduni-nin**
Add an option `--check-links` to the verify command, which tests all URL given in the manifest.

As a result, it would be convenient to have a way to check the links in the manifests in order to identify typos and outdated links.

Resolves: #11

### CODE REVIEW

File: lib/verifier.rb [hunk 1]
```diff
@@ -120,4 +120,54 @@ def verify_file filename
...
+    end
+    if manifest.urls.announcement
+      verify_link('announcement',manifest.urls.announcement)
+    end
+    if manifest.urls.mailing_list
+      verify_link('mailing_list',manifest.urls.mailing_list)
+    end
+    if manifest.urls.contact
+      verify_link('contact',manifest.urls.contact)
+    end
```
Comment (cornelius) [inline lib/verifier.rb]: This is quite repetitive code. This can be written with less duplication.

File: lib/verifier.rb [hunk 2]
```diff
@@ -120,4 +120,54 @@ def verify_file filename
...
+    puts ""
+  end
+
+  def verify_link name, link
+    url = URI.parse(link)
+    req = Net::HTTP.new(url.host, url.port)
+
+    if link.include? "https"
```
Comment (cornelius) [inline lib/verifier.rb]: This does not check the protocol, but a random occurrence of `https` in the URL. This should be more explicit. Maybe it also would be better to use some way which transparently does that for you.

File: lib/verifier.rb [hunk 3]
```diff
@@ -120,4 +120,54 @@ def verify_file filename
...
+      req.use_ssl = true
+    end
+    res = req.request_head(url.path)
+
+    if res.code == "200"
```
Comment (cornelius) [inline lib/verifier.rb]: What happens when the link redirects somewhere else? Is this reported as an error?

**Comment by cornelius** [comment]
This is nice for checking the links in one manifest.

It also would be nice to have a way to check all links in all manifests, at least in the latest version of the manifest, which is shown as link on the web site.

File: lib/verifier.rb [hunk 4]
```diff
@@ -120,4 +120,54 @@ def verify_file filename
...
+    if manifest.urls.announcement
+      verify_link('announcement',manifest.urls.announcement)
+    end
+    if manifest.urls.mailing_list
+      verify_link('mailing_list',manifest.urls.mailing_list)
+    end
+    if manifest.urls.contact
+      verify_link('contact',manifest.urls.contact)
+    end
+    puts ""
```
Comment (cornelius) [inline lib/verifier.rb]: Just `puts` does the same.

**Comment by cornelius** [comment]
Please don't force push to an active pull request. This makes it hard to follow what has changed. Just adding another commit addressing the comments is much easier to understand. When merging commits can still be squashed if needed.

**Comment by cornelius** [comment]
Did you address the comments somehow? I don't see related changes.

**Comment by nanduni-nin** [comment]
I am still working on finding some means to address redirecting cases.

File: lib/verifier.rb (hunk 1 above)
Comment (nanduni-nin) [inline lib/verifier.rb]: I am not sure if you are meaning removing this type of a duplication.
url = manifest.urls.contact
verify_link('contact',url) if !url.nil?

Comment (cornelius) [inline lib/verifier.rb]: I mean something like replacing the nine distinct if clauses by a loop and only having one if clause inside the loop.
